File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  const confirmLength = nums.length / 2;
  const numsObj = {};
  
  for (const element of nums) {
    if (numsObj[element]) {
      numsObj[element]++;
    } else {
      numsObj[element] = 1;
    }

    if (numsObj[element] > confirmLength) {
      return element;
    }
  }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash_map = {}

        for num in nums:
            if num in hash_map: 
                hash_map[num] += 1
            else:
                hash_map[num] = 1

            if hash_map[num] >= len(nums) / 2:
                return num
